public/app/features/dashboard/history/history.html

```diff
@@ -0,0 +1,161 @@
+							<tbody>
+								<tr ng-repeat="revision in ctrl.revisions">
+									<td bs-tooltip="ctrl.compareRevisionDisabled(revision.checked) ? 'You can only compare 2 versions at a time' : ''">
+										<gf-form-switch
+											checked="revision.checked"
+											on-change="ctrl.compareRevisionStateChanged(revision)"
+											ng-disabled="ctrl.compareRevisionDisabled(revision.checked)">
+										</gf-form-switch>
+									</td>
+									<td>{{revision.version}}</td>
+									<td>{{ctrl.formatDate(revision.created)}}</td>
+									<td>{{revision.createdBy}}</td>
+									<td>{{revision.message}}</td>
+									<td class="text-right">
+										<a class="btn btn-inverse btn-small" ng-show="revision.version !== ctrl.dashboard.version" ng-click="ctrl.restore(revision.version)">
+											<i class="fa fa-rotate-right"></i>&nbsp;&nbsp;Restore
+										</a>
+										<a class="btn btn-outline-disabled btn-small" ng-show="revision.version === ctrl.dashboard.version">
+											<i class="fa fa-check"></i>&nbsp;&nbsp;Current
+										</a>
+									</td>
+								</tr>
+							</tbody>
+						</table>
+
+					<div ng-if="ctrl.appending">
+						<i class="fa fa-spinner fa-spin"></i>
+						<em>Fetching more entries&hellip;</em>
+					</div>
+
+						<div class="gf-form-group" ng-show="ctrl.mode === 'list'">
+							<div class="gf-form-button-row">
+								<a	type="button"
+										class="btn gf-form-button btn-primary"
+										ng-if="ctrl.revisions.length > 1"
+										ng-class="{disabled: !ctrl.isComparable()}"
+										ng-click="ctrl.getDiff(ctrl.diff)"
+										bs-tooltip="ctrl.isComparable() ? '' : 'Select 2 versions to start comparing'">
+									<i class="fa fa-code-fork" ></i>&nbsp;&nbsp;Compare versions
+								</a>
+								<a  type="button"
+										class="btn gf-form-button btn-inverse"
+										ng-if="ctrl.revisions.length >= ctrl.limit"
+										ng-click="ctrl.addToLog()"
+										ng-class="{disabled: ctrl.isLastPage()}"
+										ng-disabled="ctrl.isLastPage()">
+									Show more versions
+								</a>
+							</div>
+						</div>
+					</div>
+				</div>
+			</div>
+		</div>
+
+		<div class="history-list" ng-if="ctrl.mode === 'compare'">
+			<div class="page-container">
+				<div class="page-body">
+					<aside class="page-sidebar">
+						<section class="page-sidebar-section">
+							<ul class="ui-list">
+								<li><a ng-class="{active: ctrl.diff === 'basic'}" ng-click="ctrl.getDiff('basic')" href="">Change Summary</a></li>
+								<li><a ng-class="{active: ctrl.diff === 'html'}" ng-click="ctrl.getDiff('html')" href="">JSON Code View</a></li>
+							</ul>
+						</section>
+					</aside>
+
+					<div class="tab-content page-content-with-sidebar">
+						<div ng-if="ctrl.loading">
+							<i class="fa fa-spinner fa-spin"></i>
+							<em>Fetching changes&hellip;</em>
+						</div>
+
+						<div ng-if="!ctrl.loading" ng-init="new = ctrl.selected[0]; original = ctrl.selected[1]">
```

Comment:
trying to stay away from using ng-init and creating variables in the "scope" directly from the view, creates messy templates & logic that can be hard to maintain.